### PR TITLE
chore: support getting multiple endpoints from the `Provision` rpc call

### LINKED ISSUE #7973: allow multiple SideroLink endpoints
Allow multiple Wireguard endpoints in response to `Provision()` request.

### PR DESCRIPTION
**Pull request by DmitriyMV**
Closes #7973

### CODE REVIEW

File: internal/app/machined/pkg/controllers/siderolink/manager.go [hunk 1]
```diff
@@ -246,6 +246,13 @@ func (ctrl *ManagerController) Run(ctx context.Context, r controller.Runtime, lo
 		linkSpec := network.NewLinkSpec(network.ConfigNamespaceName, network.LayeredID(network.ConfigOperator, network.LinkID(constants.SideroLinkName)))
 		addressSpec := network.NewAddressSpec(network.ConfigNamespaceName, network.LayeredID(network.ConfigOperator, network.AddressID(constants.SideroLinkName, nodeAddress)))
 
+		endpoints := resp.GetEndpoints()
```
Comment (smira) [inline internal/app/machined/pkg/controllers/siderolink/manager.go:249]: this is not that easy... we should support cycling through the endpoints if the connection is down (today we just do full reconnect, but we should try all endpoints first)

File: internal/app/machined/pkg/controllers/siderolink/manager.go [hunk 2]
```diff
@@ -146,106 +148,129 @@ func (ctrl *ManagerController) Run(ctx context.Context, r controller.Runtime, lo
...
+		if len(endpoints) == 0 {
+			return errors.New("host returned no endpoints")
+		}
+
+		// Rotate through the endpoints.
+		ep := endpoints[ctrl.endpointIdx]
+		ctrl.endpointIdx = (ctrl.endpointIdx + 1) % len(endpoints)
+
+		logger.Info(
+			"configurating siderolink connection",
```
Comment (smira) [inline internal/app/machined/pkg/controllers/siderolink/manager.go:268]:
```suggestion
			"configuring siderolink connection",
```

File: internal/app/machined/pkg/controllers/siderolink/manager.go (hunk 1 above)
Comment (DmitriyMV) [reply internal/app/machined/pkg/controllers/siderolink/manager.go]: Hmm, I tried approaching this by storing the endpoints response in the controller and rotating through them until we reach the 0 idx, in which case we try the provisioning again.

**Comment by DmitriyMV** [comment]
The code is now readable :)

**Review by smira — APPROVED** [review]

**Pull request merged**